spec/jobs/hyrax/grant_read_job_spec.rb

```diff
@@ -1,11 +1,28 @@
 RSpec.describe Hyrax::GrantReadJob do
   let(:depositor) { create(:user) }
-  let(:file_set) { build(:file_set) }
+  let(:file_set) { create(:file_set, user: depositor) }
 
-  it 'grants a user read access to a FileSet' do
-    expect(FileSet).to receive(:find).with(file_set.id).and_return(file_set)
-    expect(file_set).to receive(:read_users=).with(array_including(depositor.user_key))
-    expect(file_set).to receive(:save!)
-    described_class.perform_now(file_set.id, depositor.user_key)
+  context "when use_valkyrie is not set" do
+    it 'grants a user read access to a FileSet' do
+      described_class.perform_now(file_set.id, depositor.user_key)
+      file_set.reload
+      expect(file_set.read_users).to include depositor.user_key
+    end
+  end
+
+  context "when use_valkyrie is false" do
+    it 'grants a user read access to a FileSet' do
+      described_class.perform_now(file_set.id, depositor.user_key, use_valkyrie: false)
```

Comment:
To be a pure valkyrie test, create a valkyrie file_set using...
```
let(:valk_file_set) { valkyrie_create(:hydra_file_set) }
```
and then in the `perform_now` use `valk_file_set.id`

Reply:
`let(:valk_file_set) { valkyrie_create(:hyrax_file_set) }` works for me.